OpenSim/Common/Test/testDataTable.cpp

```diff
@@ -0,0 +1,134 @@
+        col_index.upd().push_back(SimTK::Value<unsigned>{i});
+
+    DataTable::DependentsMetaData dep_metadata{};
+    dep_metadata.setValueArrayForKey("labels", labels);
+    dep_metadata.setValueArrayForKey("column-index", col_index);
+
+    DataTable::IndependentMetaData ind_metadata{};
+    ind_metadata.setValueForKey("labels", std::string{"0"});
+    ind_metadata.setValueForKey("column-index", unsigned{0});
+
+    TimeSeriesTable table{};
+    table.setDependentsMetaData(dep_metadata);
+    table.setIndependentMetaData(ind_metadata);
+
+    SimTK::RowVector_<double> row{5, double{0}};
+
+    for(unsigned i = 0; i < 5; ++i)
+        table.appendRow(0.00 + 0.25 * i, row + i);
+
+    for(unsigned i = 0; i < 5; ++i)
+        table.updRowAtIndex(i) += 1;
+
+    for(unsigned i = 0; i < 5; ++i)
+        table.updRow(0 + 0.25 * i) -= 1;
+
+    try {
+        table.appendRow(0.5, row);
+    } catch (OpenSim::Exception&) {}
+
+    table.updTableMetaData().setValueForKey("DataRate", 600);
+    table.updTableMetaData().setValueForKey("Filename", 
+                                            std::string{"/path/to/file"});
+
+    // Retrieve added metadata and rows to check.
+    if(table.getNumRows() != unsigned{5})
+        throw Exception{"Test Failed: table.getNumRows() != unsigned{5}"};
+
+    if(table.getNumColumns() != unsigned{5})
+        throw Exception{"Test Failed: table.getNumColumns() != unsigned{5}"};
+
+    const auto& dep_metadata_ref = table.getDependentsMetaData();
+
+    const auto& labels_ref = dep_metadata_ref.getValueArrayForKey("labels");
+    for(unsigned i = 0; i < 5; ++i)
+        if(labels_ref[i].getValue<std::string>() != std::to_string(i + 1))
+            throw Exception{"Test failed: labels_ref[i].getValue<std::string>()"
+                    " != std::to_string(i + 1)"};
+
+    const auto& col_index_ref 
+        = dep_metadata_ref.getValueArrayForKey("column-index");
+    for(unsigned i = 0; i < 5; ++i)
+        if(col_index_ref[i].getValue<unsigned>() != i + 1)
+            throw Exception{"Test failed: col_index_ref[i].getValue<unsigned>()"
+                    " != i + 1"};
+
+    const auto& ind_metadata_ref = table.getIndependentMetaData();
+    
+    if(ind_metadata_ref.getValueForKey("labels").getValue<std::string>() 
+       != std::string{"0"})
+        throw Exception{"Test failed: ind_metadata_ref.getValueForKey"
+                "(\"labels\").getValue<std::string>() != std::string{\"0\"}"};
+    if(ind_metadata_ref.getValueForKey("column-index").getValue<unsigned>()
+       != unsigned{0})
+        throw Exception{"Test failed: ind_metadata_ref.getValueForKey"
+                "(\"column-index\").getValue<unsigned>() != unsigned{0}"};
+
+    for(unsigned i = 0; i < 5; ++i) {
+        for(unsigned j = 0; j < 5; ++j) {
+            const auto row_i_1 = table.getRowAtIndex(i);
+            if(row_i_1[j] != (row + i)[j])
+                throw Exception{"Test failed: row_i_1[j] != (row + i)[j]"};
```

Comment:
Now that you have the exception macros (and if there are no objections to them) you could use them. I don't think it is necessary for the test case but would be nice to see them in use.